### PR TITLE
Long commit with build date replaced

### PR DESCRIPTION
**Pull request by lazzy-panda**
Long commit with build date replaced

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [ ] No need to
- [ ] Manually (please, describe, if necessary)
- [x] Unit checks
- [ ] Integration checks
- [x] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by Haarolean — CHANGES_REQUESTED** [review]
I observe no date unfortunately
image

**Review by Haarolean — CHANGES_REQUESTED** [review]
the date belongs to release date, not build time/commit time

File: kafka-ui-react-app/src/components/Version/Version.tsx [hunk 1]
```diff
@@ -24,14 +26,15 @@ const Version: React.FC<VesionProps> = ({ tag, commit }) => {
         setLatestVersionInfo({
           outdated: compareVersions(tag, data.tag_name) === -1,
           latestTag: data.tag_name,
+          date: dayjs(data.published_at).format('MM.DD.YYYY HH:mm:ss'),
```
[the remark this answers is not in the document]
Comment (Haarolean) [reply kafka-ui-react-app/src/components/Version/Version.tsx:29]: we need time as well but it should be build time, not a latest release datetime

[the remark this answers is not in the document]
Comment (lazzy-panda) [reply kafka-ui-react-app/src/components/Version/Version.tsx]: I added last build date with time

File: kafka-ui-react-app/src/components/Version/__tests__/Version.spec.tsx [hunk 2]
```diff
@@ -9,14 +9,8 @@ const commit = '123sdf34';
 describe('Version', () => {
   const setupComponent = (props: VesionProps) => render(<Version {...props} />);
 
-  it('renders', () => {
-    setupComponent({ tag });
-    expect(screen.getByText(tag)).toBeInTheDocument();
-  });
-
-  it('shows current tag and commit', () => {
+  it('shows current commit', () => {
```
[the remark this answers is not in the document]
Comment (lazzy-panda) [reply kafka-ui-react-app/src/components/Version/__tests__/Version.spec.tsx]: added

File: kafka-ui-react-app/package.json [hunk 3]
```diff
@@ -47,9 +47,9 @@
     ]
   },
   "scripts": {
-    "start": "react-scripts start",
+    "start": "export REACT_APP_COMMIT=$(git rev-parse --short HEAD) && export REACT_APP_BUILD_DATE=$(date '+%F %H:%M:%S') && react-scripts start",
```
Comment (workshur) [inline kafka-ui-react-app/package.json:50]: I'm not sure it the correct place for such logic

File: kafka-ui-react-app/src/components/Version/__tests__/Version.spec.tsx [hunk 4]
```diff
@@ -2,21 +2,20 @@ import React from 'react';
...
-
-  it('shows current tag and commit', () => {
-    setupComponent({ tag, commit });
-    expect(screen.getByText(tag)).toBeInTheDocument();
+  it('shows latest commit hash and date', () => {
+    setupComponent({ tag, commit, date });
     expect(screen.getByText(commit)).toBeInTheDocument();
+    expect(
+      screen.getByText(dayjs(date).format('MM.DD.YYYY HH:mm:ss'))
+    ).toBeInTheDocument();
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Version/__tests__/Version.spec.tsx:19]: I want to see case when tag, commit, date are undefined

File: kafka-ui-react-app/package.json (hunk 3 above)
Comment (lazzy-panda) [reply kafka-ui-react-app/package.json:50]: This is the most common way to do this task. There is another option to transfer this logic to webpack. But if we use vite now, I can investigate the question of how to move it there. Should I do this?

File: kafka-ui-react-app/src/components/Version/__tests__/Version.spec.tsx (hunk 4 above)
Comment (lazzy-panda) [reply kafka-ui-react-app/src/components/Version/__tests__/Version.spec.tsx:19]: done

File: kafka-ui-react-app/package.json (hunk 3 above)
Comment (workshur) [reply kafka-ui-react-app/package.json:50]: How did it work before?

**Review by workshur — CHANGES_REQUESTED** [review]

File: kafka-ui-react-app/package.json (hunk 3 above)
Comment (lazzy-panda) [reply kafka-ui-react-app/package.json:50]: We got some information from the request to backend:
https://api.github.com/repos/provectus/kafka-ui/releases/latest

But now there is no information we need there

Comment (Haarolean) [reply kafka-ui-react-app/package.json:50]: > How did it work before?

before that there was no build time, and we have to either parse it on the frontend part or to pass it via npm build, like we already do with the tag

Comment (workshur) [reply kafka-ui-react-app/package.json:50]: https://github.com/provectus/kafka-ui/blob/1b71ccb9755041e18ff540d5c57318243b38876d/kafka-ui-api/pom.xml#L404-L405

Comment (workshur) [reply kafka-ui-react-app/package.json:50]: I think all this info should be provided on build step
https://github.com/git-commit-id/git-commit-id-maven-plugin/blob/master/docs/using-the-plugin-in-more-depth.md

**Pull request closed without merging**